src/listmode_buildblock/CListRecordROOT.cxx

```diff
@@ -50,6 +51,8 @@ void CListEventROOT::get_detection_position(DetectionPositionPair<>& _det_pos) c
 
     _det_pos.pos1() = det1;
     _det_pos.pos2() = det2;
+    _det_pos.timing_pos() = this->get_uncompressed_proj_data_info_sptr()->get_tof_bin(delta_time);
```

Comment:
This is a highly dangerous line. 
I remembered why I had the unmashed version in the previous commit. 
If you set mashing factor in the hroot file, then the get_uncompressed_proj_data_info_sptr() is mashed. 
That is alright if you want to do LM reconstruction with mashing factor.

However, if you do lm_to_projdata you want always you LM data unmashed. 
The line bellow ensures that your LM data will always come unmashed, even if you have forgotten a mashing factor. 

However, if your LM data from a scanner come naturally mashed, then you need to either: (a) have a scanner template that is mashed or (b) use the line above. Otherwise, both LM reconstruction and lm_to_projdata might fail. 

For GATE the best is to always return the unmashed and let the template in the LmToProjdata decide the mashing.

Reply:
I'd have to check this, but the name `get_uncompressed_proj_data_info_sptr()` implies that no mashing should have happened. If it did for `ROOT`, it might be a bug there?

Reply:
It is not a bug really. It is a conflict between the TOF mashing applied in this point, which is needed when listmode data are TOF compressed (in comparison to the scanner template). And the TOF mashing applied at ```ProjDataInfoCylindricalNoArcCorr::get_bin_for_det_pos_pair(Bin& bin, const DetectionPositionPair<>& dp)```. These two applications conflict in some scenarios. E.g. The Scanner template doesn't have the TOF mashing factor that has been applied in the listmode data, but the user applies that in the hroot file, which would be alright for listmode reconstruction. However if with the same setup make a sinogram with a template that contains TOF mashing then the final sinogram will be double compressed.